src/main/java/duke/TaskList.java

```diff
@@ -74,18 +74,45 @@ public void removeTask(int index) {
      * @param t The task to be saved.
      */
     public void saveTask(Task t) {
-        Taskmanager.storage.write(t.getSavingFormat());
+        TaskMaster.storage.write(t.getSavingFormat());
+    }
+
+    /**
+     * Finds a task from the ArrayList based on a key.
```

Comment:
ok but i think it can be phrased better, i had to read the function to figure out what you meant by key, maybe call it a substring?